### ISSUE TITLE
Implementing a version of the (Sparse) Categorical Cross Entropy Loss that makes use of Integer-Indexed predictions.

### ISSUE THREAD

**Issue by mayowaosibodu** [keras-team/keras#14506]
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

The Categorical Cross Entropy Loss makes use of both one-hot encoded predictions, and one-hot encoded true labels.
Eg, y_true = [ [0,0,1], [1,0,0] ] and y_pred = [ [0,0,1], [0,1,0] ].

The Sparse Categorical Cross Entropy Loss makes use of single integer true labels, but still requires an array of probabilities corresponding to each existing class, per label in the predictions.
Eg, y_true = [2, 0] and y_pred = [ [0.1, 0.2, 0.7], [0.2, 0.85, 0.05] ].

I feel it would be really useful to have a loss function which makes use of single integer predictions as well as single integer true labels.
Eg, y_true = [2, 0] and y_pred = [2, 1].

**Will this change the current api? How?**
Not really. It could possibly just require the definition of a new loss function.

**Who will benefit with this feature?**
For one, it would be possible to use Dense models to process two-dimensional data. This could be done by simply encoding one of the concerned dimensions as a series of integer-indexed categories.

Eg you could process text data with a Dense model by simply encoding the characters as integer-indexed categories (in contrast to one-hot encoding) and then work on minimizing the outputs of the proposed loss function.

Eg, "I am" would simply be [9,0,1,13] (with ' ' as 0). This is evidently amenable to processing by a Dense model.

**Comment by sushreebarsa** [comment]
@mayowaosibodu 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
